libraries/WiFi/src/WiFiScan.cpp

```diff
@@ -117,8 +117,13 @@ int16_t
  */
 void WiFiScanClass::_scanDone() {
   esp_wifi_scan_get_ap_num(&(WiFiScanClass::_scanCount));
+  if (WiFiScanClass::_scanResult) {
+    delete[] reinterpret_cast<wifi_ap_record_t *>(WiFiScanClass::_scanResult);
+    WiFiScanClass::_scanResult = 0;
```

Comment:
using `NULL` or `nullptr` here is preferable (even though it all resolves to `0`)